users/src/commonMain/kotlin/majestic/users/profile/roles/index.kt

```diff
@@ -0,0 +1,274 @@
+package majestic.users.profile.roles
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.unit.dp
+import captain.Navigator
+import composex.screen.orientation.Landscape
+import composex.screen.orientation.Portrait
+import composex.screen.orientation.ScreenOrientation
+import majestic.ColorPair
+import majestic.ExpandDirection
+import majestic.FloatingActionButton
+import majestic.floatActionButton
+import majestic.tooling.onClick
+import majestic.users.labels.roles.RolesLabels
+import majestic.users.labels.settings.LanguageController
+import majestic.users.labels.settings.observeUsersLabels
+import majestic.users.profile.roles.assignRoleModalContent.AssignRoleModalContent
+import majestic.users.profile.roles.campus.Campus
+import majestic.users.profile.roles.campus.CampusMenuAction
+import majestic.users.tools.ProfilePortraitHeader
+import majestic.users.tools.buttons.ButtonAnimate
+import majestic.users.tools.buttons.FlatButton
+import majestic.users.tools.dialogs.Modal
+import tz.co.asoft.majestic_users.generated.resources.Res
+import tz.co.asoft.majestic_users.generated.resources.ic_add
+
+@Composable
+private fun RolesHeader(
+    title: String,
+    colors: RolesColors,
+    orientation: ScreenOrientation
+) {
+    if (orientation is Landscape) {
+        Text(
+            modifier = Modifier.padding(vertical = 20.dp, horizontal = 30.dp),
+            text = title,
+            color = colors.theme.surface.contra.color.copy(0.5f),
+        )
+    }
+}
+
+@Composable
+private fun CampusesList(
+    campuses: List<CampusData>,
+    labels: RolesLabels,
+    colors: RolesColors,
+    orientation: ScreenOrientation,
+    onCampusAction: (String, CampusMenuAction) -> Unit
+) {
+    campuses.forEach { campus ->
+        Campus(
+            campusName = campus.name,
+            rolesCount = campus.rolesCount,
+            labels = labels,
+            colors = colors.campus,
+            orientation = orientation,
+            onAddRole = { onCampusAction(campus.id, CampusMenuAction.AddRole) },
+            onViewRole = { onCampusAction(campus.id, CampusMenuAction.ViewRole) },
+            onEditRole = { onCampusAction(campus.id, CampusMenuAction.EditRole) },
+            onDeleteRole = { onCampusAction(campus.id, CampusMenuAction.DeleteRole) }
+        )
+    }
+}
+
+@Composable
+private fun LandscapeAddButton(
+    label: String,
+    colors: majestic.users.tools.buttons.ButtonAnimateColors,
+    isOpen: Boolean,
+    onToggle: () -> Unit
+) {
+    Box(modifier = Modifier.offset(y = (-40).dp, x = (-30).dp)) {
```

Comment:
Move modifier definition to the call site. I understand why you need a box for this. So, don't bust your head here to try and achieve this without a Box. But if you can, that would be an impressive show of skillset